log/file.go

```diff
@@ -44,15 +44,19 @@ type fileHook struct {
 	w              io.WriteCloser
 	bw             *bufio.Writer
 	levels         []logrus.Level
+	done           chan struct{}
 }
 
 // FileHookFromConfigLine returns new fileHook hook.
 func FileHookFromConfigLine(
-	ctx context.Context, fallbackLogger logrus.FieldLogger, line string,
+	ctx context.Context, fallbackLogger logrus.FieldLogger, line string, done chan struct{},
```

Comment:
This done channel looks more like the internal state. Do you think we should accept it as input?

Can a signature be like?

```go
FileHookFromConfigLine(ctx context.Context, fallbackLogger logrus.FieldLogger, line string) (logrus.Hook, chan struct{},error)`
```

Wdyt?

Reply:
Ideally, even with the doneCh `<-chan sruct{},` but I guess it is outside of the scope of this PR

Reply:
hmm yeah, that is slightly better :+1:

Reply:
actually, I think I'll refactor both this and `LokiFromConfigLine` in a separate PR, so I'll leave it as it is for now